File: leetcode/completed/2033.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        int m = grid.size(), n = grid[0].size();
        vector<int> flatten;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                flatten.push_back(grid[i][j]);
            }
        }
        sort(flatten.begin(), flatten.end());

        int target = flatten[flatten.size()/2];

        int res = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (abs(grid[i][j]-target) % x > 0) return -1;
                res += abs(grid[i][j]-target) / x;
            }
        }

        return res;

    }
};

File: leetcode/completed/2033.java
public class 2033 {
    class Solution {
        public int minOperations(int[][] grid, int x) {
            int m = grid.length, n = grid[0].length;
            int[] flatten = new int[m*n];
    
            for (int i = 0; i < m; i++) {
                for (int j = 0; j < n; j++ ) {
                    flatten[i*n+j] = grid[i][j];
                }
            }
    
            Arrays.sort(flatten);
            int target = flatten[flatten.length / 2];
    
            int res = 0;
            for (int i = 0; i < m; i++) {
                for (int j = 0; j < n; j++ ) {
                    if (Math.abs(grid[i][j] - target) % x != 0) return -1;
                    res += Math.abs(grid[i][j] - target) / x;
                }
            }
    
            return res;
    
        }
    }
}
